### PR TITLE
Install dependencies when starting a new development stack

### PR DESCRIPTION
**Pull request by Leodau**
## What does this PR do ?

Makes the _run-dev.sh_ script which is executed inside docker, install all dependencies (core & plugins) if they haven't been installed yet.

### How should this be manually tested?

To run the stack, you would only need to clone and launch with docker-compose:

`git clone -b feature/automatic-dependencies-install https://github.com/kuzzleio/kuzzle.git && cd kuzzle && docker-compose up`

### CODE REVIEW

**Comment by Aschen** [comment]
Oh was is still WIP?

File: docker-compose/scripts/run-dev.sh [hunk 1]
```diff
@@ -8,11 +8,23 @@ fi
 
 elastic_host=${kuzzle_services__storageEngine__client__node:-http://elasticsearch:9200}
 
+if [ ! -d "./node_modules/" ]; then
+    git submodule init
+    git submodule update
+    npm ci --unsafe-perm
+    ./docker-compose/scripts/install-plugins.sh
+fi
```
Comment (scottinet) [inline docker-compose/scripts/run-dev.sh:16]: These lines seem redundant: the `elif` condition below seems sufficient. What's their added value?

**Comment by scottinet** [comment]
I've taken the liberty of changing the PR title (it's used as our changes history so it needs to be explicit), and the label: since this only affects development stacks, I believe that this shouldn't appear in our changelogs.

Let me know if you disagree.

File: docker-compose/scripts/run-dev.sh (hunk 1 above)
Comment (Leodau) [reply docker-compose/scripts/run-dev.sh:16]: Yep :+1: , test/old chunk went through. Fixed

**Pull request merged**